plugins/woocommerce/src/Blocks/Domain/Services/Schema/DocumentObject.php

```diff
@@ -0,0 +1,165 @@
+<?php
+declare( strict_types = 1);
+
+namespace Automattic\WooCommerce\Blocks\Domain\Services\Schema;
+
+use WC_Cart;
+use WC_Customer;
+use Automattic\WooCommerce\StoreApi\StoreApi;
+use Automattic\WooCommerce\StoreApi\SchemaController;
+use Automattic\WooCommerce\StoreApi\Schemas\V1\CartSchema;
+use Automattic\WooCommerce\StoreApi\Schemas\V1\BillingAddressSchema;
+use Automattic\WooCommerce\StoreApi\Schemas\V1\ShippingAddressSchema;
+use Automattic\WooCommerce\StoreApi\Utilities\LocalPickupUtils;
+use Automattic\WooCommerce\StoreApi\Utilities\CartController;
+
+/**
+ * DocumentObject class.
+ *
+ * This will combine and format given cart/customer/checkout data into a standard object format that can be queried through
+ * JSON. This is used for conditional fields and validation during checkout.
+ */
+class DocumentObject {
+	/**
+	 * The cart object.
+	 *
+	 * @var WC_Cart|null
+	 */
+	protected $cart = null;
+
+	/**
+	 * The customer object.
+	 *
+	 * @var WC_Customer|null
+	 */
+	protected $customer = null;
+
+	/**
+	 * Cart controller class instance.
+	 *
+	 * @var CartController
+	 */
+	protected $cart_controller;
+
+	/**
+	 * Schema controller class instance.
+	 *
+	 * @var SchemaController
+	 */
+	protected $schema_controller;
+
+	/**
+	 * The request data.
+	 *
+	 * @var array
+	 */
+	protected $request_data = [];
+
+	/**
+	 * The constructor.
+	 *
+	 * @param array $request_data Data that overrides the default values.
+	 */
+	public function __construct( array $request_data = [] ) {
+		$this->cart_controller   = new CartController();
+		$this->schema_controller = StoreApi::container()->get( SchemaController::class );
+		$this->cart              = $this->cart_controller->get_cart_for_response();
```

Comment:
Should we be getting cart this early? I think it's not copying so it's fine?

Reply:
We could do it later (related to https://github.com/woocommerce/woocommerce/pull/54851#discussion_r1933643553) but for now its fine. Its private anyway.

Reply:
I've moved them in https://github.com/woocommerce/woocommerce/pull/54851/commits/7c2a037028994da0ab0a1869f78a65fae7b45ce9 with a note.